ckanext/canada/view.py

```diff
@@ -579,6 +625,64 @@ def ckanadmin_publish_datasets():
     return h.redirect_to('canada.ckanadmin_publish')
 
 
+@canada_views.route('/ckan-admin/job-queue', methods=['GET'])
+def ckanadmin_job_queue():
+    """
+    Lists all of the queued and running jobs.
+    """
+    try:
+        jobs = get_action('job_list')({}, {})
+    except NotAuthorized:
+        abort(403, _('Not authorized to see this page'))
+
+    warning = False
+    if jobs and datetime.strptime(jobs[0]['created'], '%Y-%m-%dT%H:%M:%S') < (datetime.now() - timedelta(minutes=18)):
+        warning = True
+
+    job_list = []
+    for job in jobs[:MAX_JOB_QUEUE_LIST_SIZE]:
+        try:
+            job_obj = Job.fetch(job.get('id'))
```

Comment:
Can `job_list` be extended to return the information you need instead of querying the back end from a view function?

Reply:
@wardi all of this logic from the view method has now been moved into a chained logic action for `job_list`.